tests/validation/tests/v3_api/common.py

```diff
@@ -535,21 +562,40 @@ def wait_until_active(url, timeout=120):
     return
 
 
-def validate_http_response(cmd, target_name_list, client_pod=None):
-    if client_pod is None and cmd.startswith("http://"):
+def validate_http_response(cmd, target_name_list):
+    if cmd.startswith("http://"):
         wait_until_active(cmd, 60)
     target_hit_list = target_name_list[:]
     count = 5 * len(target_name_list)
     for i in range(1, count):
         if len(target_hit_list) == 0:
             break
-        if client_pod is None:
-            curl_cmd = "curl " + cmd
-            result = run_command(curl_cmd)
-        else:
-            wget_cmd = "wget -qO- " + cmd
-            result = kubectl_pod_exec(client_pod, wget_cmd)
-            result = result.decode()
+        curl_cmd = "curl " + cmd
+        result = run_command(curl_cmd)
+        result = result.rstrip()
+        print("cmd: \t" + cmd)
+        print("result: \t" + result)
+        assert result in target_name_list
+        if result in target_hit_list:
+            target_hit_list.remove(result)
+    print("After removing all, the rest is: ", target_hit_list)
+    assert len(target_hit_list) == 0
+
+
+def validate_http_response_inside_pod(cmd, target_name_list, client_pod):
```

Comment:
I think this is a copy test method of `validate_http_response` and added support for windows pod to make http request. Can we determine the pod os type by scheduling rule, like `beta.kubernetes.io/os = windows` means windows? So we can still use the `validate_http_response` method instead of creating a separate one.

Reply:
I am going to modify this and test again.